src/OpenTelemetry.Contrib.Instrumentation.Quartz/TraceProviderBuilderExtensions.cs

```diff
@@ -0,0 +1,48 @@
+// <copyright file="TraceProviderBuilderExtensions.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using OpenTelemetry.Contrib.Instrumentation.Quartz;
+using OpenTelemetry.Contrib.Instrumentation.Quartz.Implementation;
+
+// ReSharper disable once CheckNamespace
+namespace OpenTelemetry.Trace
+{
+    /// <summary>
+    /// Extension methods to simplify registering of dependency instrumentation.
+    /// </summary>
+    public static class TraceProviderBuilderExtensions
+    {
+        /// <summary>
+        /// Enables the Quartz.NET Job automatic data collection for Quartz.NET.
+        /// </summary>
+        /// <param name="builder"><see cref="TraceProviderBuilderExtensions"/> being configured.</param>
+        /// <param name="configureQuartzInstrumentationOptions">Quartz configuration options.</param>
+        /// <returns>The instance of <see cref="TraceProviderBuilderExtensions"/> to chain the calls.</returns>
+        public static TracerProviderBuilder AddQuartzInstrumentation(
+            this TracerProviderBuilder builder,
+            Action<QuartzInstrumentationOptions> configureQuartzInstrumentationOptions = null)
+        {
+            var options = new QuartzInstrumentationOptions();
+            configureQuartzInstrumentationOptions?.Invoke(options);
+
+            builder.AddInstrumentation(() => new QuartzJobInstrumentation(options));
+            builder.AddSource(QuartzDiagnosticListener.ActivitySourceName);
```

Comment:
does Quartz library publish DiagnosticSource events? Could you point to the source code where it is doing so? I'd assume its already creating Activity, using legacy way, and hence we need to run them through samplers, etc, by using AddLegacySource API..
https://github.com/open-telemetry/opentelemetry-dotnet/tree/main/docs/trace/extending-the-sdk#special-case--instrumentation-for-libraries-producing-legacy-activity

Reply:
Quartz  library publishes DiagnosticSource events:
https://github.com/quartznet/quartznet/blob/main/src/Quartz/Logging/JobDiagnosticsWriter.cs

Reply:
Okay so it does start the Activity. I doubt if the activity actually goes through the samplers, as you are not calling `AddLegacySource` for this activity.

Reply:
cc: @utpilla

Reply:
On Quartz side we can also change things per recommendations how the current state of things should be. The current beta-based implementation has been marketed as unstable API etc so happy to steer towards ideal solution.

Reply:
if that is a possibility,  I strongly recommend that :) 
https://github.com/open-telemetry/opentelemetry-dotnet/tree/main/docs/trace/extending-the-sdk#writing-own-instrumentation-library

It'd involve using ActivitySource model of creating activities. Once the Quartz library does that, then *this* instrumentation is optional. :)

Reply:
I'll let @maldago steer this, I don't have enough subject expertise and grateful for him doing the effort 👍🏻